rust/numaflow-core/src/pipeline.rs

```
@@ -125,6 +136,35 @@ pub(crate) async fn start_forwarder(
             )
             .await?;
         }
+        pipeline::VertexType::Reduce(reduce) => {
+            info!("Starting reduce forwarder");
+
+            // create watermark handle, if watermark is enabled
+            let edge_watermark_handle = match &config.watermark_config {
+                Some(WatermarkConfig::Edge(edge_config)) => Some(
+                    ISBWatermarkHandle::new(
+                        config.vertex_name,
+                        config.replica,
+                        config.read_timeout,
+                        js_context.clone(),
+                        edge_config,
+                        &config.to_vertex_config,
+                        cln_token.clone(),
+                    )
+                    .await?,
+                ),
+                _ => None,
```

this should return an error since reduce can't progress without watermark?